### PR TITLE
Add a class to manipulate a DynamoDB

### PR DESCRIPTION
**Pull request by adanaja**
Ignore the warning B905 as the strict option has been added to zip in Python3.10 only

### CODE REVIEW

File: src/e3/aws/dynamodb/__init__.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        logger.info(f"Retrievieng item {item} from {table_name}...")
+        try:
+            response = table.get_item(
+                Key={key: item[key] for key in keys if key in item.keys()}
+            )
+            logger.debug(f"Get_item response: {response}")
+        except ClientError as e:
+            logger.error(e)
+            return {}
+        else:
```
Comment (grouigrokon) [inline src/e3/aws/dynamodb/__init__.py:84]: Using a `finally` here would helps having a single `return` statement, but that is not needed. Just a remark.

File: src/e3/aws/dynamodb/__init__.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            attr["IndexName"] = index_name
+
+        paging = True
+        items = []
+        while paging:
+            result = table.query(**attr)
+            items += result["Items"]
+            if "LastEvaluatedKey" in result:
+                # we have not scanned all table
+                attr.update({"ExclusiveStartKey": result["LastEvaluatedKey"]})
```
Comment (grouigrokon) [inline src/e3/aws/dynamodb/__init__.py:184]: The `attr` variable is local, and does not seem to be used after that line.

Comment (adanaja) [reply src/e3/aws/dynamodb/__init__.py:184]: It's a while loop

Comment (grouigrokon) [reply src/e3/aws/dynamodb/__init__.py:184]: Sorry, I was misleaded by the `(**attr)` (basically I simply missed it !)

File: src/e3/aws/dynamodb/__init__.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        table_name: str,
+        query: dict[str, list[str]],
+        sort_key: tuple[str, str] | None = None,
+        index_name: str | None = None,
+    ) -> list[dict[str, Any]]:
+        """Query items from a table.
+
+        :param table_name: name of the table
+        :param query: a dict containing a key value pair for the query
+            e.g. {"issue": "eng/repo1#1"}
```
Comment (pierretr) [inline src/e3/aws/dynamodb/__init__.py:143]: This example could probably be more generic and here it's not a `dict[str, list[str]]` that the function query_items method expects.

Comment (adanaja) [reply src/e3/aws/dynamodb/__init__.py]: Updated

**Review by pierretr — DISMISSED** [review]

**Review by pierretr — APPROVED** [review]

**Pull request merged**